scripts/compile/compile-corehost.ps1

```diff
@@ -5,42 +5,158 @@
+            mkdir $HostDir | Out-Null
+        }
+        cp "$RepoRoot\src\corehost\cmake\$Rid\cli\$BuildConfiguration\corehost.exe" $HostDir
+        cp "$RepoRoot\src\corehost\cmake\$Rid\cli\dll\$BuildConfiguration\hostpolicy.dll" $HostDir
+    
+        if (Test-Path "$RepoRoot\src\corehost\cmake\$Rid\cli\$BuildConfiguration\corehost.pdb")
+        {
+            cp "$RepoRoot\src\corehost\cmake\$Rid\cli\$BuildConfiguration\corehost.pdb" $HostDir
+        }
+        if (Test-Path "$RepoRoot\src\corehost\cmake\$Rid\cli\dll\$BuildConfiguration\hostpolicy.pdb")
+        {
+            cp "$RepoRoot\src\corehost\cmake\$Rid\cli\dll\$BuildConfiguration\hostpolicy.pdb" $HostDir
+        }
+
+        Package-Host
+    } finally {
+        popd
     }
-    $BuildConfiguration = $Configuration
-    if ($Configuration -eq "Release") {
-        $BuildConfiguration = "RelWithDebInfo"
+}
+
+function Package-Host
+{
+    $RidPrefix = "runtime.$RID"
+    $PackageName = $CoreHostPackageName
+    $PackageVersion=[System.IO.File]::ReadAllText("$RepoRoot\src\corehost\packaging\.version").Trim()
+
+    $BinFiles = @{
+        'ID'="$RidPrefix.$PackageName"
+        'JSON'=$False
+        'FILES'= @"
+    <file src="corehost.exe" target="$CoreHostRelativeDir\corehost.exe"></file>
+    <file src="hostpolicy.dll" target="$CoreHostRelativeDir\hostpolicy.dll"></file>
+"@
     }
-    & "$pf\MSBuild\14.0\Bin\MSBuild.exe" ALL_BUILD.vcxproj /p:Configuration="$BuildConfiguration"
-    if (!$?) {
-        Write-Host "Command failed: $pf\MSBuild\14.0\Bin\MSBuild.exe" ALL_BUILD.vcxproj /p:Configuration="$BuildConfiguration"
-        Exit 1
+
+    $PdbFiles = @{
+        'ID'="$RidPrefix.$PackageName.Symbols"
+        'JSON'=$False
+        'FILES'= @"
+    <file src="corehost.pdb" target="$CoreHostRelativeDir\corehost.pdb"></file>
+    <file src="hostpolicy.pdb" target="$CoreHostRelativeDir\hostpolicy.pdb"></file>
+"@
     }
 
-    if (!(Test-Path $HostDir)) {
-        mkdir $HostDir | Out-Null
+    $BinJsonFile = @{
+        'ID'="$PackageName"
+        'JSON'=$True
+        'FILES'= @"
+    <file src="$PackageName.runtime.json" target="runtime.json"></file>
+"@
     }
-    cp "$RepoRoot\src\corehost\cmake\$Rid\cli\$BuildConfiguration\corehost.exe" $HostDir
-    cp "$RepoRoot\src\corehost\cmake\$Rid\cli\dll\$BuildConfiguration\hostpolicy.dll" $HostDir
 
-    if (Test-Path "$RepoRoot\src\corehost\cmake\$Rid\cli\$BuildConfiguration\corehost.pdb")
-    {
-        cp "$RepoRoot\src\corehost\cmake\$Rid\cli\$BuildConfiguration\corehost.pdb" $HostDir
+    $PdbJsonFile = @{
+        'ID'="$PackageName.Symbols"
+        'JSON'=$True
+        'FILES'= @"
+    <file src="$PackageName.Symbols.runtime.json" target="runtime.json"></file>
+"@
     }
-    if (Test-Path "$RepoRoot\src\corehost\cmake\$Rid\cli\dll\$BuildConfiguration\hostpolicy.pdb")
-    {
-        cp "$RepoRoot\src\corehost\cmake\$Rid\cli\dll\$BuildConfiguration\hostpolicy.pdb" $HostDir
+
+    $NuSpecContents = @"
```

Comment:
To reduce the overhead of updating versions or file tags in multiple files, esp. Unix.